java/ql/src/semmle/code/java/security/QueryInjection.qll

```diff
@@ -0,0 +1,48 @@
+/** Provides classes to reason about database query language injection vulnerabilities. */
+
+import java
+import semmle.code.java.dataflow.DataFlow
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.frameworks.android.SQLite
+import semmle.code.java.frameworks.javaee.Persistence
+import semmle.code.java.frameworks.SpringJdbc
+import semmle.code.java.frameworks.MyBatis
+import semmle.code.java.frameworks.Hibernate
+
+/** A sink for database query language injection vulnerabilities. */
+abstract class QueryInjectionSink extends DataFlow::ExprNode { }
```

Comment:
```suggestion
abstract class QueryInjectionSink extends DataFlow::Node { }
```
Since we're opening this up for more open-ended extension, we shouldn't restrict ourselves to `ExprNode`s. This change also requires a few `this.getExpr()` below to be changed to `this.asExpr()`.